### PR TITLE
Add deleteWithRetry for secgroups

### PR DESCRIPTION
**Pull request by lego963**

### CODE REVIEW

File: openstack/networking/v2/extensions/security/groups/requests.go [hunk 1]
```diff
@@ -123,6 +123,20 @@ func Delete(c *golangsdk.ServiceClient, id string) (r DeleteResult) {
 	return
 }
 
+// DeleteWithRetry will try to permanently delete a particular security
+// group based on its unique ID and RetryTimeout.
+func DeleteWithRetry(c *golangsdk.ServiceClient, id string, timeout int) error {
+	return golangsdk.WaitFor(timeout, func() (bool, error) {
+		_, err := c.Delete(resourceURL(c, id), nil)
+		if err != nil {
+			if _, ok := err.(golangsdk.ErrDefault404); ok {
```
Comment (outcatcher) [inline openstack/networking/v2/extensions/security/groups/requests.go:132]: Please wait not for 404 iteration later, but for no error at all

**Review by outcatcher — CHANGES_REQUESTED** [review]

File: openstack/compute/v2/extensions/secgroups/requests.go [hunk 2]
```diff
@@ -100,6 +100,20 @@ func Delete(client *golangsdk.ServiceClient, id string) (r DeleteResult) {
 	return
 }
 
+// DeleteWithRetry will try to permanently delete a particular security
+// group based on its unique ID and RetryTimeout.
+func DeleteWithRetry(c *golangsdk.ServiceClient, id string, timeout int) error {
+	return golangsdk.WaitFor(timeout, func() (bool, error) {
+		_, err := c.Delete(resourceURL(c, id), nil)
```
[the remark this answers is not in the document]
Comment (lego963) [reply openstack/compute/v2/extensions/secgroups/requests.go:107]: we don't have implementation of polling interval in `.WaitFor()` func, should I do this one?
image

[the remark this answers is not in the document]
Comment (outcatcher) [reply openstack/compute/v2/extensions/secgroups/requests.go]: There is 1 second retry inteerval built-in into `WaitFor`, but I assume it is still too oft

File: openstack/compute/v2/extensions/secgroups/requests.go [hunk 3]
```diff
@@ -100,6 +100,21 @@ func Delete(client *golangsdk.ServiceClient, id string) (r DeleteResult) {
...
 
+// DeleteWithRetry will try to permanently delete a particular security
+// group based on its unique ID and RetryTimeout.
+func DeleteWithRetry(c *golangsdk.ServiceClient, id string, timeout int) error {
+	return golangsdk.WaitFor(timeout, func() (bool, error) {
+		response, err := c.Delete(resourceURL(c, id), nil)
+		if err != nil {
+			return false, err
+		}
+		if response.StatusCode == 202 || response.StatusCode == 204 {
```
Comment (outcatcher) [inline openstack/compute/v2/extensions/secgroups/requests.go:111]: No need to check those statuses, they are checked in `ProviderClient.Request`

File: openstack/networking/v2/extensions/security/groups/requests.go [hunk 4]
```diff
@@ -123,6 +123,18 @@ func Delete(c *golangsdk.ServiceClient, id string) (r DeleteResult) {
...
 
+// DeleteWithRetry will try to permanently delete a particular security
+// group based on its unique ID and RetryTimeout.
+func DeleteWithRetry(c *golangsdk.ServiceClient, id string, timeout int) error {
+	return golangsdk.WaitFor(timeout, func() (bool, error) {
+		_, err := c.Delete(resourceURL(c, id), nil)
+		if err == nil {
+			return true, nil
+		}
+		return false, nil
```
Comment (outcatcher) [inline openstack/networking/v2/extensions/security/groups/requests.go:134]: You should return `nil` err only in case of HTTP error

**Review by anton-sidelnikov — APPROVED** [review]

**Review by outcatcher — APPROVED** [review]

**Pull request merged**